core/domain/html_cleaner.py

```diff
@@ -203,7 +246,14 @@ def convert_to_text_angular(html_data):
     # of other tags which produces issues in migration.
     html_data = html_data.replace('<br>', '<br/>')
 
-    soup = bs4.BeautifulSoup(html_data, 'html.parser')
+    # This is to check whether a collapsible or tabs component is present
+    # in html_data. If that is the case migrate the content inside those
+    # components using convert_tag_contents_to_text_angular().
+    if any(component in html_data for component in [
```

Comment:
There are edge cases where this willl incorrectly evaluate true. For example if the content contains the literal text "oppia-noninteractive-collapsible" but not the actual tag.

If there are no collapsible or tabs, `convert_tag_contents_to_text_angular` should just do nothing right? I'd say simply remove the if-check and add comments both here and in the docstring of `convert_tag_contents_to_textangular` specifying that the function converts content inside tabs or collapsible, *if those components exist*, otherwise it does nothing.

Reply:
Yeah! right.